[Doc] Editorial review to fields (advanced) (#4686)
taichi-dev/taichi: docs/lang/articles/advanced/layout.md

By nesting multiple statements, we can construct a field with higher dimensions.
<!-- haidong: how far can we go? how many default axis exist? -->

In order to traverse the nested statements, you can use `struct-for`:

```python {1}
for i, j in A:
    A[i, j] += 1
```

> The order to access `A`, namely the order to iterate `i` and `j`, affects the program performance subtly. The Taichi compiler is capable o f automatically deducing the underlying data layout and applying a proper access order. This is an advantage over most general-purpose programming languages where the access order has to be optimized manually.

Reviewer: ```suggestion
The order to access `A`, namely the order to iterate `i` and `j`, affects the program performance subtly. The Taichi compiler is capable of automatically deducing the underlying data layout and applying a proper access order. This is an advantage over most general-purpose programming languages where the access order has to be optimized manually.
```

> 
> ...

Reviewer: ```suggestion
```

    pos[i] += vel[i] * dt
    vel[i] += -k * pos[i] * dt
```

Merely revising the place statement is sufficient to change the layout. With this optimization, the instant elements `pos[i]` and `vel[i]` are now adjacent in memory, which is more efficient.

<!-- ```python

> 

Reviewer: ```suggestion
```

# SoA version

> 

Reviewer: ```suggestion
```

def step():
    pos[i] += vel[i] * dt
    vel[i] += -k * pos[i] * dt

> 

Reviewer: ```suggestion
```

<!-- ```python
ti.root.dense(ti.i, 3).place(x)
ti.root.dense(ti.i, 3).place(y)

> 

Reviewer: ```suggestion
```

> ...

Reviewer: ```suggestion
```
